controllers/profiles/ensurer.go

```diff
@@ -24,34 +24,48 @@ import (
 	operatorapi "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
 )
 
-// newObjectEnsurer see objectEnsurer
-func newObjectEnsurer(client client.Client, logger *logr.Logger, creator objectCreator) *objectEnsurer {
-	return &objectEnsurer{
+type ObjectEnsurer interface {
+	ensure(ctx context.Context, workflow *operatorapi.KogitoServerlessWorkflow, visitors ...mutateVisitor) (client.Object, controllerutil.OperationResult, error)
+}
+
+// newDefualtObjectEnsurer see defaultObjectEnsurer
+func newDefualtObjectEnsurer(client client.Client, logger *logr.Logger, creator objectCreator) ObjectEnsurer {
+	return &defaultObjectEnsurer{
 		client:  client,
 		logger:  logger,
 		creator: creator,
 	}
 }
 
-// objectEnsurer provides the engine for a ReconciliationState that needs to create or update a given Kubernetes object during the reconciliation cycle.
-type objectEnsurer struct {
+// defaultObjectEnsurer provides the engine for a ReconciliationState that needs to create or update a given Kubernetes object during the reconciliation cycle.
+type defaultObjectEnsurer struct {
 	client  client.Client
 	logger  *logr.Logger
 	creator objectCreator
 }
 
+// newDummyObjectEnsurer see dummyObjectEnsurer
+func newDummyObjectEnsurer() ObjectEnsurer {
+	return &dummyObjectEnsurer{}
+}
+
+// dummyObjectEnsurer is a useful Object ensurer to apply the null pattern. Use it when you need a creator that does nothing
+type dummyObjectEnsurer struct {
```

Comment:
what is the use case of it?

Reply:
This is used to implement the null-pattern, and so when we need an object ensurer that does nothing.
This is useful to avoid conditional code like Ricardo suggested